File: SEP3_Database/src/main/java/com/example/SEP3_Database/model/domain/list/postList/CommentList.java
package com.example.SEP3_Database.model.domain.list.postList;

import com.example.SEP3_Database.model.domain.unit.post.Comment;

import java.util.ArrayList;

public class CommentList {
    private ArrayList<Comment> Comments;

    public CommentList()
    {
        Comments = new ArrayList<Comment>();
    }

    public CommentList(ArrayList<Comment> commentList)
    {
        Comments = new ArrayList<Comment>();
        for (int x=0;x<commentList.size();x++)
        {
            Comments.add(commentList.get(x).copy());
        }
    }

    public boolean hasCommentId(String id)
    {
        for (int x=0;x<Comments.size();x++)
        {
            if (Comments.get(x).getCommentId()==id)
            {
                return true;
            }
        }
        return false;
    }

    public String addComment(Comment comment)
    {
        if (comment!=null)
        {
            if (!hasCommentId(comment.getCommentId()))
            {
                Comments.add(comment);
            }
            else
            {
                return "Not a new comment.";
            }
        }
        return "Input null.";
    }

    public Comment getCommentByCommentId(String id)
    {
        for (int x=0;x<Comments.size();x++)
        {
            if (Comments.get(x).getCommentId()==id)
            {
                return Comments.get(x);
            }
        }
        return null;
    }

    public Comment getCommentByIndex(int index)
    {
        if (index>=0&&index<Comments.size())
        {
            return Comments.get(index);
        }

        return null;
    }

    public void removeCommentByCommentId(String id)
    {
        for (int x=0;x<Comments.size();x++)
        {
            if (Comments.get(x).getCommentId()==id)
            {
                Comments.remove(x);
                break;
            }
        }
    }

    public CommentList copy()
    {
        return new CommentList(Comments);
    }
}

File: SEP3_PostSystem/Model/List/PostList/CommentList.cs
using System.Collections.Generic;
using System.Linq;
using SEP3_PostSystem.Model.Unit.Post;

namespace SEP3_PostSystem.Model.List.PostList
{
    public class CommentList
    {
        public List<Comment> Comments { get; set; }

        public CommentList()
        {
            Comments = new List<Comment>();
        }

        public CommentList(List<Comment> commentList)
        {
            Comments = new List<Comment>();
            foreach (var comment in commentList)
            {
                Comments.Add(comment.Copy());
            }
        }

        public bool HasCommentId(string id)
        {
            return Comments.Any(comment => comment.CommentId == id);
        }

        public string AddComment(Comment comment)
        {
            if (comment!=null)
            {
                if (!HasCommentId(comment.CommentId))
                {
                    Comments.Add(comment);
                    return null;
                }
                else
                {
                    return "Not a new comment.";
                }
            }
            return "Input null.";
        }

        public Comment GetCommentByCommentId(string id)
        {
            foreach (var comment in Comments)
            {
                if (comment.CommentId==id)
                {
                    return comment;
                }
            }
            return null;
        }

        public Comment GetCommentByIndex(int index)
        {
            if (index>=0&&index<Comments.Count)
            {
                return Comments[index];
            }

            return null;
        }

        public void RemoveCommentByCommentId(string id)
        {
            foreach (var comment in Comments.Where(comment => comment.CommentId==id))
            {
                Comments.Remove(comment);
                break;
            }
        }

        public CommentList Copy()
        {
            return new CommentList(Comments);
        }
    }
}